File: dialogs/devices/consts.py
import enum


class QueryError(enum.Enum):
    DeviceUnreachable = 'DEVICE_UNREACHABLE'
    DeviceBusy = 'DEVICE_BUSY'
    DeviceNotFound = 'DEVICE_NOT_FOUND'
    InternalError = 'INTERNAL_ERROR'


class ActionError(enum.Enum):
    DeviceUnreachable = 'DEVICE_UNREACHABLE'
    DeviceBusy = 'DEVICE_BUSY'
    DeviceNotFound = 'DEVICE_NOT_FOUND'
    InternalError = 'INTERNAL_ERROR'
    InvalidAction = 'INVALID_ACTION'
    InvalidValue = 'INVALID_VALUE'
    NotSupportedInCurrentMode = 'NOT_SUPPORTED_IN_CURRENT_MODE'


class ActionStatus(enum.Enum):
    Done = 'DONE'
    Error = 'ERROR'

File: src/protocol/consts.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum ActionError {
    DeviceUnreachable,
    DeviceBusy,
    DeviceNotFound,
    InternalError,
    InvalidAction,
    InvalidValue,
    NotSupportedInCurrentMode,
}

impl ActionError {
    pub fn as_str(self) -> &'static str {
        match self {
            ActionError::DeviceUnreachable => "DEVICE_UNREACHABLE",
            ActionError::DeviceBusy => "DEVICE_BUSY",
            ActionError::DeviceNotFound => "DEVICE_NOT_FOUND",
            ActionError::InternalError => "INTERNAL_ERROR",
            ActionError::InvalidAction => "INVALID_ACTION",
            ActionError::InvalidValue => "INVALID_VALUE",
            ActionError::NotSupportedInCurrentMode => "NOT_SUPPORTED_IN_CURRENT_MODE",
        }
    }
}

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum ActionStatus {
    Done,
    Error,
}

impl ActionStatus {
    pub fn as_str(self) -> &'static str {
        match self {
            ActionStatus::Done => "DONE",
            ActionStatus::Error => "ERROR",
        }
    }
}
